keps/sig-network/5311-relaxed-validation-for-service-names/README.md

```diff
@@ -0,0 +1,425 @@
+- a search in the Kubernetes bug triage tool (https://storage.googleapis.com/k8s-triage/index.html)
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+If e2e tests are not necessary or useful, explain why.
+-->
+
+- Create a Service that requires the new validation and test if a DNS lookup works for it
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- E2E and Integration tests completed and enabled.
+
+#### GA
+
+- Time passes, no major objections
+- Promote e2e test to conformance
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+### Version Skew Strategy
+
+Not applicable - only a single component is being changed.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: RelaxedServiceNameValidation
+  - Components depending on the feature gate:
+    - kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+No, as this feature is for validation only.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, via the feature gate.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The new relaxed validation will be enabled again.
+
+###### Are there any tests for feature enablement/disablement?
+
+The integration test will test disabling of the feature
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+An initial rollout cannot fail because the feature only changes the
+behavior when creating new Services and doesn't affect already-created
+Services.
+
+A rollback will always succeed if the user hasn't created any Services that depend
+on the new validation.
+
+A rollback to a version that has the feature gate will work, even if there are
+Services that depend on the new validation, since any version of Kubernetes with
+this feature gate will allow existing Services to have `NameIsDNSLabel` names. If a
+user rolls back to a version before the feature gate, and has Services that depend
+on the new validation, they will be unable to modify those services, and will
```

Comment:
This not entirely true, is it? We only validate service when creating, so theoretically they should function. One won't be able to create the services which rely on relaxed validation? Right?

Reply:
No, currently `ValidateServiceCreate` and `ValidateServiceUpdate` use a shared `ValidateService` function that mostly doesn't pay attention to the distinction between creates and updates.,